### PR TITLE
Improve Table performance

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

Improve the Table render performance. You can find detailed information about the metrics generated in this improvement [here](https://www.notion.so/vtexhandbook/Table-Performance-d85396c61a4f4b448b7a7e60b9c14ca1)

#### Breaking Changes
- Table composites are now required to implement the Table component
- `onRowClick` is defined using the `TBodyRow` onClick property
- `useTableState` doesn't receives the `useDataView` state return anymore, now it only receives the status
- `getRowKey` doesn't exist anymore, since now the user have full control of the item

**Before**

```tsx
import { Table, createColumns, useTableState } from '@vtex/admin-ui'

const columns = createColumns(...)
const items = [...]

function Example() {
  const state = useTableState({
    columns,
    items,
  })

  return <Table state={state} />
}
```

**After**

```tsx
import {
  Table,
  THead,
  THeadCell,
  TBody,
  TBodyRow,
  TBodyCell,
  createColumns,
  useTableState
} from '@vtex/admin-ui'

const columns = createColumns(...)
const items = [...]

function Example() {
  const { getTable, getHeadCell, getBodyCell, data } = useTableState({
    columns,
    items,
  })

  return (
    <Table {...getTable()}>
      <THead>
        {columns.map((column) => {
          <THeadCell {...getHeadCell(column)} />
        })
      </THead>
      <TBody>
        {items.map((item) => (
            <TBodyRow>
              {columns.map((column) => (
                <TBodyCell {...getBodyCell(column, item)} />
              ))}
            </TBodyRow>
        )}
      </TBody>
    </Table>
  )
}
```

#### How should this be manually tested?

You can compare the Table stories from this branch and from the dev branch.

#### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Quality improvement (tests or refactors)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/table/components/styles/table-row.styles.ts [hunk 1]
```diff
@@ -2,4 +2,5 @@ import { style } from '@vtex/admin-ui-core'
 
 export const rowBaseline = style({
   display: 'contents',
+  position: 'absolute',
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/styles/table-row.styles.ts:5]:
```suggestion
```

File: packages/admin-ui/src/table/components/table-body/table-body-cell.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      lastFixedColumn={lastFixedColumn}
+    >
+      {content}
+    </TableCell>
+  )
+}
+
+export interface TableBodyCellProps<T> extends TableCellProps<T> {
+  item: T
+  status: DataViewStatus
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/table-body/table-body-cell.tsx:32]:
```suggestion
```

File: packages/admin-ui/src/table/components/table-body/table-body-row.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import type { StyleProp } from '@vtex/admin-ui-core'
+import React, { memo } from 'react'
+import { Box } from '../../../box'
+
+import * as styles from '../styles/table-body.styles'
+
+function TableBodyRow(props: TableBodyRowProps) {
+  const { selected, children, onClick, ...rowProps } = props
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/table-body/table-body-row.tsx:8]:
```suggestion
  const { selected, children, onClick, csx, ...rowProps } = props
```

File: packages/admin-ui/src/table/components/table-body/table-body.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React, { memo } from 'react'
+import type { StyleProp } from '@vtex/admin-ui-core'
+
+import { Box } from '../../../box'
+
+import * as styles from '../styles/table-body.styles'
+
+function TableBody(props: TableBodyProps) {
+  const { children, ...restProps } = props
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/table-body/table-body.tsx:9]:
```suggestion
  const { children, csx, ...restProps } = props
```

File: packages/admin-ui/src/table/components/table-head/table-head-cell.tsx [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  sortState: UseSortReturn
+  key: React.Key
+}
+
+function SortIndicator(props: BoxProps & SortIndicatorOptions) {
+  const { direction, ...restProps } = props
+
+  return (
+    <Box csx={styles.sortIndicator} {...restProps}>
+      <Fragment>
```
Comment (lucasaarcoverde) [inline packages/admin-ui/src/table/components/table-head/table-head-cell.tsx:94]:
```suggestion
```

File: packages/admin-ui/src/table/hooks/use-table-state.tsx [hunk 6]
```diff
@@ -111,22 +91,65 @@ export function useTableState<T>(
...
+  )
+
+  const getTable = useCallback(
+    () => ({
+      columns,
+      status,
+      tableRef,
+    }),
+    [columns, status]
+  ) as any
```
Comment (anitavincent) [inline packages/admin-ui/src/table/hooks/use-table-state.tsx:139]: Could it be "as () => TableProps<?>"?

File: packages/admin-ui/src/table/stories/complexity.stories.tsx [hunk 7]
```diff
@@ -54,60 +54,70 @@ const columns = createColumns<Item>([
...
 
-export function Zero() {
-  const state = useTableState<Item>({
-    columns,
-    items,
-  })
-
-  return <Table state={state} csx={{ width: 560 }} />
-}
+// const Example = React.memo(() => {
```
Comment (anitavincent) [inline packages/admin-ui/src/table/stories/complexity.stories.tsx:57]: Remove this?

**Review by anitavincent — APPROVED** [review]

**Review by Lenderson — APPROVED** [review]

**Pull request merged**